### PR TITLE
`asserting in`

### PR DESCRIPTION
**Pull request by dewert99**
Requires viperproject/silver#660
This is my attempt at adding supporting the `asserting in` feature in Carbon.
The "tests/all/asserting-in/wand.vpr" test from silver fails. Although, when I was initially working on this, this was also an issue when encoding `asserting in` using function preconditions, now the error seems specific to `asserting in`, so it may be worth looking into.

### CODE REVIEW

File: silver
Comment (alexanderjsummers) [inline silver:1]: I'm not sure how pull requests work across submodules; could this be a separate pull request for silver?

Comment (alexanderjsummers) [reply silver:1]: In any case, github doesn't seem to understand how to present the diffs for these files.

File: src/main/scala/viper/carbon/modules/impls/DefaultExpModule.scala [hunk 1]
```diff
@@ -74,9 +74,7 @@ class DefaultExpModule(val verifier: Verifier) extends ExpModule with Definednes
         sys.error("should not occur here (either, we inhale or exhale this expression, in which case whenInhaling/whenExhaling should be used, or the expression is not allowed to occur.")
       case p@sil.PredicateAccess(_, _) =>
         translateLocationAccess(p)
-      case sil.Unfolding(_, exp) =>
-        translateExp(exp)
-      case sil.Applying(_, exp) => translateExp(exp)
+      case h: sil.HintExp => translateExp(h.body)
```
Comment (alexanderjsummers) [inline src/main/scala/viper/carbon/modules/impls/DefaultExpModule.scala:77]: This is nice to have. Is HintExp a sealed trait?

**Review by alexanderjsummers — CHANGES_REQUESTED** [review]
I think the pull request needs splitting into a silver one and a carbon one.

File: silver
Comment (dewert99) [reply silver:1]: I did make two pull requests: see https://github.com/viperproject/silver/pull/660, this pull request references the silver commit in the silver pull request

**Comment by jcp19** [comment]
I will close this, given that #531 was already merged.

**Pull request closed without merging**